sdk/metricsadvisor/Azure.AI.MetricsAdvisor/api/Azure.AI.MetricsAdvisor.netstandard2.0.cs

```diff
@@ -768,6 +768,11 @@ public abstract partial class DataFeedSource
         public static bool operator !=(Azure.AI.MetricsAdvisor.Models.DataFeedStatus left, Azure.AI.MetricsAdvisor.Models.DataFeedStatus right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class DataLakeGen2SharedKeyDatasourceCredential : Azure.AI.MetricsAdvisor.Models.DatasourceCredential
```

Comment:
I imagine there is a reason why the name needs to be super specific?

Reply:
Data Lake Gen 2 is the name of the service to which this credential is used. Shared Key specifies the type of credential. We will need to create other datasource credentials for different types of authentication.